var/spack/repos/builtin/packages/glew/package.py

```diff
@@ -75,3 +75,8 @@ def cmake_args(self):
             args.append(self.define("OPENGL_egl_LIBRARY", "IGNORE"))
 
         return args
+
+    if sys.platform == "darwin":
+
+        def setup_build_environment(self, env):
+            env.prepend_path("LDFLAGS", "-framework OpenGL")
```

Comment:
Could you implement this as a `flag_handler`:

```
    def flag_handler(self, name, flags):
        if name == "ldflags" and self.spec.satisfies("platform=darwin"):
            flags.append("-framework OpenGL")
        return (None, None, flags)
```

? Using that approach can improve mixing of flags with flags in e.g. compilers.yaml.

> I'm compiling with gl=other so I'm not completely sure if when using different variants with macOS this change is needed or not.

I think this in fact should condition on that or more specifically on `^apple-gl` like:

```
    def flag_handler(self, name, flags):
        if name == "ldflags" and self.spec.satisfies("platform=darwin ^apple-gl"):
        ...
```

Reply:
Done!